tools/linux_vsts.ps1

```diff
@@ -0,0 +1,300 @@
+    }
+}
+
+function Invoke-PSBootstrap {
+    Write-Host -Foreground Green "Executing Linux vsts -BootStrap `$isPR='$isPr' - $commitMessage"
+    # Make sure we have all the tags
+    Sync-PSTags -AddRemoteIfMissing
+    Start-PSBootstrap -Package:$createPackages
+}
+
+function Invoke-PSBuild {
+    $releaseTag = Get-ReleaseTag
+
+    Write-Host -Foreground Green "Executing Linux vsts `$isPR='$isPr' `$isFullBuild='$isFullBuild' - $commitMessage"
+    $originalProgressPreference = $ProgressPreference
+    $ProgressPreference = 'SilentlyContinue'
+    try {
+        Start-PSBuild -CrossGen -PSModuleRestore -CI -ReleaseTag $releaseTag -Configuration 'Release'
+        Save-PSOptions
+    }
+    finally{
+        $ProgressPreference = $originalProgressPreference
+    }
+}
+
+function Invoke-PSTest {
+    Restore-PSOptions
+    $testResultsNoSudo = "$pwd/TestResultsNoSudo.xml"
+    $testResultsSudo = "$pwd/TestResultsSudo.xml"
+    $output = Split-Path -Parent (Get-PSOutput -Options (New-PSOptions))
+    $pesterParam = @{
+        'binDir'         = $output
+        'PassThru'       = $true
+        'Terse'          = $true
+        'Tag'            = @()
+        'ExcludeTag'     = @('RequireSudoOnUnix')
+        'OutputFile'     = $testResultsNoSudo
+        'ThrowOnFailure' = $true
+    }
+
+    if ($isFullBuild) {
+        $pesterParam['Tag'] = @('CI','Feature','Scenario')
+    } else {
+        $pesterParam['Tag'] = @('CI')
+    }
+
+    if ($hasRunFailingTestTag)
+    {
+        $pesterParam['IncludeFailingTest'] = $true
+    }
+
+    # Remove telemetry semaphore file in CI
+    $telemetrySemaphoreFilepath = Join-Path $output DELETE_ME_TO_DISABLE_CONSOLEHOST_TELEMETRY
+    if ( Test-Path "${telemetrySemaphoreFilepath}" ) {
+        Remove-Item -force ${telemetrySemaphoreFilepath}
+    }
+
+    # Running tests which do not require sudo.
+    $pesterPassThruNoSudoObject = Start-PSPester @pesterParam
+
+    # Running tests, which require sudo.
+    $pesterParam['Tag'] = @('RequireSudoOnUnix')
+    $pesterParam['ExcludeTag'] = @()
+    $pesterParam['Sudo'] = $true
+    $pesterParam['OutputFile'] = $testResultsSudo
+    $pesterPassThruSudoObject = Start-PSPester @pesterParam
+
+    # Determine whether the build passed
+    try {
+        @($pesterPassThruNoSudoObject) | ForEach-Object { Test-PSPesterResults -ResultObject $_ }
+    }
+    catch {}
+
+    try {
+        $SequentialXUnitTestResultsFile = "$pwd/SequentialXUnitTestResults.xml"
+        $ParallelXUnitTestResultsFile = "$pwd/ParallelXUnitTestResults.xml"
+
+        Start-PSxUnit -SequentialTestResultsFile $SequentialXUnitTestResultsFile -ParallelTestResultsFile $ParallelXUnitTestResultsFile
+        # If there are failures, Test-XUnitTestResults throws
+        $SequentialXUnitTestResultsFile, $ParallelXUnitTestResultsFile | ForEach-Object { Test-XUnitTestResults -TestResultsFile $_ }
+    }
+    catch {}
```

Comment:
why the empty catch block? If it is needed, therere should be a comment or the exception being logged.